src/corehost/cli/roll_forward_option.h

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef __ROLL_FORWARD_OPTION_H_
+#define __ROLL_FORWARD_OPTION_H_
+
+// Specifies the roll forward option value
+enum class roll_forward_option
+{
+    // The order is in increasing level of relaxation
+    // Lower values are more restrictive than higher values
+
+    Disable = 0,    // No roll-forward is allowed - only exact match
+    LatestPatch = 1, // Roll forward to latest patch.
+    Minor = 2,       // Roll forward to closest minor but same major and then highest patch
+    LatestMinor = 3, // Roll forward to highest minor.patch but same major
+    Major = 4,       // Roll forward to closest major.minor and then highest patch
+    LatestMajor = 5, // Roll forward to highest major.minor.patch
+
```

Comment:
Two ideas.  

- Adding `Closest` prefix may help understanding.
- allowPatch = false corresponds to `ClosestPatch`

```C#

enum class roll_forward_option
{
    Disable = 0,    // No roll-forward is allowed - only exact match
    ClosestPatch = 1, // allowPatch = false
    LatestPatch = 2, // Roll forward to latest patch.
    ClosestMinor = 3,       // Roll forward to closest minor but same major and then highest patch
    LatestMinor = 4, // Roll forward to highest minor.patch but same major
    ClosestMajor = 5,       // Roll forward to closest major.minor and then highest patch
    LatestMajor = 6, // Roll forward to highest major.minor.patch
};
```

Reply:
As noted in the other comment `applyPatch=false` acts as a modifier to `LatestPatch`, `Minor` and `Major` (for backward compat reasons). Unforuntately adding it in some way into the enum would complicate the unification rules when dealing with the enum. So I chose to keep it separate. Also beacuse it's there only for backward compat reasons, so to not pollute the enum.

In general I would like to keep the enum to match the spec exactly (to match the values for the `rollForward` setting in runtime config), that way it is easy to reason about the input versus the code behavior.

I guess we could consider renaming the `Minor` -> `ClosestMinor` and `Major` -> `ClosestMajor`. I personally don't like that too much though (it's not exactly closest due to automatic path roll forward).

I guess I could add a comment describing the high-level design points for the roll-forward feature, which are:
* Try to use version which is close to what the app wanted
* Always pick the latest patch for servicing/security reasons
* Have good defaults
* Allow customizing

The first point is basically what you're trying to add into the enum itself. I do understand why, but I'm not sure it's worth it.